test/sample.cmake

```diff
@@ -0,0 +1,5 @@
+                                              FetchContent_Declare(fmt
+GIT_REPOSITORY                          https://github.com/fmtlib/fmt/
+              GIT_TAG                                           master
+                      GIT_SHALLOW True
+)
```

Comment:
Additionally, we should de-format the main `test/CMakeLists.txt` to show that the regex also picks that up.

Reply:
> One issue I see with the current approach is that cmake-format will miss any configuration files in nested directories. From the [cmake-format](https://github.com/cheshirekow/cmake_format) docs:
> 
> `` If no configuration file is specified on the command line, cmake-format will attempt to find a suitable configuration for each ``inputpath`` by checking recursively checking it's parent directory up to the root of the filesystem. ``
> 
> This would probably break with the format and check targets as we are currently copying and checking source files in the build directory.

I got rid of that logic, and instead opted for using `include`, as mentioned at https://github.com/cheshirekow/cmake_format/issues/201#issuecomment-631628511, and as can be seen at https://github.com/johelegp/jegp/commit/4dfbe8f5446ba1f53439a36ce12119fe285268c9. I haven't integrated Format.cmake into CI yet, so you can't see it working, but it works on my machine.
> 
> > Consider that I, and probably other users of both CPM and cmake-format, would like CPM to ship with a config file with its commands
> 
> Hm it's a shame that cmake-format doesn't pick up the named arguments automatically. I'll have to think about what a good solution is to distributing the config. What makes this difficult is that `CPM.cmake` forwards any additional arguments to `FetchContent`, so the command declaration should somehow "inherit" `FetchContent_Declare`'s declaration to accurately capture all possible parameters.

https://github.com/cheshirekow/cmake_format comes with additional tools. One such is `cmake-genparsers`, which I used to generate the `CPMFindPackage` output that can be glimpsed at the commit linked above. It's still limited as documented, so I had to manually add the forwarded arguments I used. But you can use `cmake-genparsers` to scan your system's FetchContent's and ExternalProject's commands to manually merge them with the output for `CPMFindPackage` and `CPMAddPackage` to make them as complete as possible.
> 
> > I'm thinking of using FetchContent to make CPM available, rather than the suggested wget (or CMake equivalent with file(DOWNLOAD) that I use) to make the module and config available and up-to-date.
> 
> A disadvantage of downloading CPM.cmake through CMake is that it would make offline configuration impossible if it isn't already in the build directory.
> 
> I see that you are grabbing the most recent version from the `master` branch. It's better to download from a specific release or commit, as any potential breaking change in CPM.cmake would silently break your configuration script otherwise.

For now, I prefer to live at head.